### ISSUE TITLE
Run forum sample tests in CI

### ISSUE THREAD

**Issue by letmaik** [microsoft/CCF#1796]
`samples/apps/forum`, essentially `npm install && npm test`. Some changes may have to be done to `samples/apps/forum/test/util.ts` to allow testing different configurations (enclave type etc).

**Comment by achamayou** [comment]
FWIW, these don't pass for me at the moment, I'm guessing because some parts of my JS toolchain are too old and don't support ES2020:

```
/data/amchamay/CCF/samples/apps/forum/src/controllers/poll.ts:109
        return this.kvTopics.get(this.kvTopicsKey) ?? [];
                                                    ^

SyntaxError: Invalid or unexpected token
    at Object.m._compile (/data/amchamay/CCF/samples/apps/forum/node_modules/ts-mocha/node_modules/ts-node/src/index.ts:439:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:1035:10)
npm ERR! Test failed.  See above for more details.
```

**Comment by letmaik** [comment]
> 
> 
> FWIW, these don't pass for me at the moment, I'm guessing because some parts of my JS toolchain are too old and don't support ES2020:
> 
> ```
> /data/amchamay/CCF/samples/apps/forum/src/controllers/poll.ts:109
>         return this.kvTopics.get(this.kvTopicsKey) ?? [];
>                                                     ^
> 
> SyntaxError: Invalid or unexpected token
>     at Object.m._compile (/data/amchamay/CCF/samples/apps/forum/node_modules/ts-mocha/node_modules/ts-node/src/index.ts:439:23)
>     at Module._extensions..js (internal/modules/cjs/loader.js:1035:10)
> npm ERR! Test failed.  See above for more details.
> ```

I put a fix up for this: https://github.com/microsoft/CCF/pull/1822. While CCF has full ES2020 support, node.js typically doesn't, so we simply target an older ES version for tests etc.

**Comment by achamayou** [comment]
@letmaik I had not realised that node.js didn't support ES2020, thanks for the fix.

**Comment by achamayou** [comment]
closed by #1826
